src/neuroglancer/volume_rendering/volume_render_layer.ts

```diff
@@ -168,22 +171,32 @@ vec4 outputColor;
 void userMain();
 `);
         defineChunkDataShaderAccess(builder, chunkFormat, numChannelDimensions, `curChunkPosition`);
+        // See Real-Time Volume Graphics, page 12
         builder.addFragmentCode(`
 void emitRGBA(vec4 rgba) {
-  float alpha = rgba.a * uBrightnessFactor;
-  outputColor += vec4(rgba.rgb * alpha, alpha);
+  float opacityCorrectedAlpha = 1.0 - (pow(clamp(1.0 - rgba.a, 0.0, 1.0), uSamplingRatio));
+  float compositedAlpha = (1.0 - outputColor.a) * opacityCorrectedAlpha;
```

Comment:
This is indeed a more accurate handling of transparency within the chunk, but the problem is that this only works if the samples can be depth ordered.  If there is more than one chunk, we can't achieve that easily.

However, currently this shader just adds up the alpha-weighted samples without regard to depth.  Instead it could do the same thing that the order-independent transparency  is doing: https://github.com/google/neuroglancer/blob/7cee81c8a60baec4263627f5394172eed0751b22/src/neuroglancer/perspective_view/panel.ts#L87

That would fix the artifacts at chunk boundaries, and would probably result in at least somewhat better normalization.

Fundamentally I think some type of gain factor may be unavoidable, but perhaps there could be a mechanism to adjust it automatically, e.g. based on the actual brightness distribution of the rendered result.  While the gain could effectively just be included in the line-segment-based transfer function, I expect users will want to control the overall brightness as they change other view parameters, and having to move multiple points in the transfer function would be inconvenient.